src/Contexts/CourseInfoProvider.js

```diff
@@ -0,0 +1,18 @@
+import React, { createContext, useState } from 'react'
+
+export const CourseInfo = createContext()
+
+export const CourseInfoProvider = ({ children }) => {
+
+    const [ courseInfo, setCourseInfo ] = useState({
+      courses: [],
+      searchURL: '',
+      currentCourse: {}
+    })
+
+    return(
+      <CourseInfo.Provider value={{ courseInfo, setCourseInfo }}>
+        {children}
```

Comment:
Good to know that this is a part of React. I did not see that `{children}` is an option in any of the documentation.